pkg/kubelet/dockertools/manager.go

```diff
@@ -948,6 +955,16 @@ func (dm *DockerManager) KillPod(pod kubecontainer.Pod) error {
 		wg.Add(1)
 		go func(container *kubecontainer.Container) {
 			defer util.HandleCrash()
+
+			// TODO: Handle this without signaling the pod infra container to
+			// adapt to the generic container runtime.
+			if container.Name == PodInfraContainerName {
+				err := dm.networkPlugin.TearDownPod(pod.Namespace, pod.Name, kubeletTypes.DockerID(container.ID))
+				if err != nil {
+					glog.Errorf("Failed tearing down the infra container: %v", err)
+					errs <- err
```

Comment:
The won't block since the channel is buffered no? (this is the same code we had before btw)